OPHD/UI/GameOptionsDialog.cpp

```diff
@@ -14,19 +14,26 @@ GameOptionsDialog::GameOptionsDialog() :
 	btnClose{"Return to Main Menu", {this, &GameOptionsDialog::onClose}}
 {
 	position({0, 0});
-	size({210, 188});
-
+	
 	const auto buttons = std::array{&btnSave, &btnLoad, &btnHelp, &btnReturn, &btnClose};
 	for (auto button : buttons)
 	{
-		button->size({200, 25});
+		button->size({buttonWidth, buttonHeight});
 	}
 
-	add(btnSave, {5, 25});
-	add(btnLoad, {5, 53});
-	add(btnHelp, {5, 81});
-	add(btnReturn, {5, 119});
-	add(btnClose, {5, 154});
+	int btnSaveHeight = buttonHeight;
+	add(btnSave, {buttonHorizontalMargin, btnSaveHeight});
+	int btnLoadHeight = btnSaveHeight + buttonHeight + buttonCloseVerticalMargin;
+	add(btnLoad, {buttonHorizontalMargin, btnLoadHeight});
+	int btnHelpHeight = btnLoadHeight + buttonHeight + buttonCloseVerticalMargin;
+	add(btnHelp, {buttonHorizontalMargin, btnHelpHeight});
+	int btnReturnHeight = btnHelpHeight + buttonHeight + buttonLargeVerticalMargin;
+	add(btnReturn, {buttonHorizontalMargin, btnReturnHeight});
+	int btnCloseHeight = btnReturnHeight + buttonHeight + buttonLargeVerticalMargin;
+	add(btnClose, {buttonHorizontalMargin, btnCloseHeight});
```

Comment:
I think this is fine, though I wanted to leave a link to some code that solved a similar problem with a data driven approach:
https://github.com/OutpostUniverse/OPHD/blob/main/OPHD/UI/ResourceBreakdownPanel.cpp#L36

Basically, it defines a data array containing parameters for each item that can't be calculated (here it's the variable margin between buttons), and then loops over that data, and calculates positions from that data.

Using that approach here might look something like:
```cpp
const std::array buttons
{
	std::tuple{btnSave, buttonCloseVerticalMargin},
	std::tuple{btnLoad, buttonCloseVerticalMargin},
	std::tuple{btnHelp, buttonCloseVerticalMargin},
	std::tuple{btnReturn, buttonLargeVerticalMargin},
	std::tuple{btnClose, buttonLargeVerticalMargin},
}

auto position = mRect.startPoint() + NAS2D::Vector{buttonHorizontalMargin, buttonHeight};
for (const auto& [button, margin] : buttons)
{
	add(button, position);
	position.y += buttonHeight + margin;
}
```

----

Of course, if we got rid of that weird spacing for the buttons, we could simplify this further. Personally I'd be in favour of removing the odd spacing between buttons. Though I think maybe the return to game button should be last, since it's basically "cancel" like behaviour that just closes the menu without doing anything. Might be better to put that somewhere easy to click, rather than a button that takes you back to the main menu, which if you didn't save, will lose all your progress.

Reply:
I'm super on board with getting rid of the differentiated spacing. If there's consensus on that, lemme know and I can refactor my commit accordingly. Thx!

Reply:
I don't know if Leeor will be around to approve, and I usually leave UI and game play decisions to him, but I would certainly approve of using consistent spacing between buttons (more so if the cancel button was last). I think the proposal is a small and low impact change that I'd be comfortable approving and merging this one myself.

Reply:
I'm game. Sorry I've been so out of touch lately. Starting to get back into it.

But yeah, don't have a lot of time to look over the PR's right now as I'm at work but if these are ready to merge I'm down.

Honestly, any improvement in the UI is fine with me.